src/Integration/Binding/IAdditionalFileReferenceChecker.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using EnvDTE;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    internal interface IAdditionalFileReferenceChecker
```

Comment:
The tests for CSharpVBProjectBinder started to get a bit complicated, so it was much easier to test this way that the class behaves correctly when both conditions are true (the additional file exists + the ruleset is referenced). The code inside is not changed.

Reply:
Testability is important, but I think the increasing number of tiny interface and implementation classes is starting to obscure the overall structure of the system e.g. it's not easy to tell which interfaces as high-level abstractions and which ones are small-scale, language specific.

Here, there's nothing to suggests that it is either CSharpVB-specific and it appears at the same level as "larger" abstractions like `IProjectBinder`, both in the namespace and in the file directory. As a minimum, it should be possible from interface name/namespace to tell that this interface is a low-level language-specific interface.

The simplest thing to do would be rename the interface/namespace. An alternative would be to nest it inside a `partial class CSharpVBProjectBinder`. Nesting the class and interface would express more clearly that it is an implementation detail of that project binder and that it isn't expected to be used elsewhere (and if it turns out to needed elsewhere, then we'd refactor).

Reply:
I'll rename to ICSharpVBAdditionalFileReferenceChecker

Reply:
The interface name should change too.